src/promql/src/planner.rs

```diff
@@ -910,9 +911,65 @@ impl PromPlanner {
             .resolve_table(table_ref.clone())
             .await
             .context(CatalogSnafu)?;
-        // Safety: `scan_filters` is not empty.
-        let result = LogicalPlanBuilder::scan(table_ref, provider, None)
+        let table_schema = provider
+            .as_any()
+            .downcast_ref::<DefaultTableSource>()
+            .context(UnknownTableSnafu)?
+            .table_provider
+            .as_any()
+            .downcast_ref::<DfTableProviderAdapter>()
+            .context(UnknownTableSnafu)?
+            .table()
+            .schema();
+
+        let time_index_type = &table_schema
+            .timestamp_column()
+            .with_context(|| TimeIndexNotFoundSnafu {
+                table: table_ref.to_quoted_string(),
+            })?
+            .data_type;
+
+        let is_time_index_ms = *time_index_type == ConcreteDataType::time_millisecond_datatype()
+            || *time_index_type == ConcreteDataType::timestamp_millisecond_datatype();
```

Comment:
Time index only allow the timestamp type.
```suggestion
        let is_time_index_ms = *time_index_type == ConcreteDataType::timestamp_millisecond_datatype();
```